crates/edr_eth/src/transaction/request/eip155.rs

```diff
@@ -121,4 +149,29 @@ mod tests {
         let request = dummy_request();
         assert_eq!(expected, request.hash());
     }
+
+    test_fake_sign_properties!();
+
+    #[test]
+    fn test_fake_sign_test_vector() -> Result<()> {
```

Comment:
In general I agree, but `anyhow::Result` is constructed in a way that's backward compatible with the standard library's `Result` and it's considered idiomatic to use it as a drop in replacement.

Reply:
From consistencies sake it doesn't make sense to make all shadowing `Result`s forbidden, except this one. It still introduces confusion about it being allowed and what `Result` type this might be, while we're gaining very little.

Reply:
Tbh I don't consider this shadowing, as `anyhow::Result` is defined as this: `pub type Result<T, E = Error> = core::result::Result<T, E>;`. So it's just the standard library's `Result` type with a default error parameter. 

For me when I see a `Result<T>`, it's obvious that it's an `anyhow::Result` since it's a common idiom in my experience (especially as a test result). 

Anyway, happy to change it if this doesn't convince you.

Reply:
The type alias still shadows or "hides" the default standard implementation, though.

Let's say we have a

```rust
mod foo {
  #[derive(thiserror::Error)]
  pub enum Error {
    ..
  }

  pub type Result<T> = core::result::Result<T, Error>;
}
```

The expectation here would be that someone uses `foo::Result<u32>` to avoid the `std::result::Result` from not being usable anymore. That's no different with `anyhow::Result` imo

Reply:
`anyhow::Result` doesn't prevent you from using `core::result::Result` since it has a default error type parameter which you can set. E.g. this works:

```rust
// Implemented as `pub type Result<T, E = Error> = core::result::Result<T, E>;`
use anyhow::Result;

#[derive(Debug, thiserror::Error)]
pub enum MyError {
    #[error("Invalid")]
    Invalid
}

fn main() {
  // Identical to `core::result::Result<(), anyhow::Error>`
  let foo: Result<()> = Ok(());
  
  // Identical to `core::result::Result<(), MyError>`
  let foo: Result<(), MyError> = Ok(());
}
```

Reply:
That's cool. I didn't know that was possible. It's the first time I've seen that form of "overloading" of types in Rust.

Would you then also accept if someone did:

```rust
use foo::Result;

fn main() {
  // Identical to `core::result::Result<(), foo::Error>`
  let foo: Result<()> = Ok(());
  
  // Identical to `core::result::Result<(), MyError>`
  let foo: Result<(), MyError> = Ok(());
}
```

I would not, as it raises confusion as to "which" `Result` type we're talking about now: is it `foo::Result`, `bar::Result`, `anyhow::Result`?

As such I'd still personally prefer usage of an explicit `anyhow::Result` in function signatures as well, since it is consistent and avoids confusion.

Reply:
> Would you then also accept if someone did:

No, it only works for me due to the prevalence of Anyhow.

> As such I'd still personally prefer usage of an explicit anyhow::Result in function signatures as well, since it is consistent and avoids confusion.

Ok done in: [Don't import anyhow::Result](https://github.com/NomicFoundation/hardhat/pull/4559/commits/09e29b69e4ca17af26884fe91c2a90896117563b)